configure.ac

```diff
@@ -1850,7 +1850,7 @@ then
         AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_AES_DIRECT"
     fi
     AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_KCAPI_AES"
-    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_KCAPI_HASH"
+    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_KCAPI_HASH -DWOLFSSL_KCAPI_HASH_KEEP"
```

Comment:
Change this to: detect if FIPS-READY is enabled before deciding to force WOLFSSL_KCAPI_HASH_KEEP.